### PR TITLE
chore: first flexible tile, and refactoring tile logic to TS

### PR DESCRIPTION
**Pull request by c-j-j**
- [ ] TileAF can be deleted - to be done in a separate PR
- [ ] We can remove the `url` from the Link component as it's not used (TODO comment left in this PR). This can be done in another PR as it'll change 100s of snapshots.

### CODE REVIEW

File: packages/edition-slices/src/flexible-tiles/tile-vertical-a/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  TileImage,
+  TileLink,
+  TileSummary,
+  withTileTracking,
+} from "../../tiles/shared";
+
+import styleFactory from "./styles";
+import WithoutWhiteSpace from "../../tiles/shared/without-white-space";
+import PositionedTileStar from "../../tiles/shared/positioned-tile-star";
+import { TransformedTile } from "@times-components-native/types";
```
Comment (aronshamash) [inline packages/edition-slices/src/flexible-tiles/tile-vertical-a/index.tsx:14]: `ConfiguredTile`  any better you think ?

Comment (aronshamash) [inline packages/edition-slices/src/flexible-tiles/tile-vertical-a/index.tsx:14]: `ConfiguredTile`  any better you think ?

File: packages/edition-slices/__tests__/ios/tile-vertical-a.test.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import shared from "../tile-vertical-a/shared-tile-vertical-a.native";
```
Comment (kamiranoff) [inline packages/edition-slices/__tests__/ios/tile-vertical-a.test.js:1]: do we still the .native file?

Comment (aronshamash) [reply packages/edition-slices/__tests__/ios/tile-vertical-a.test.js:1]: no we shouldn't

File: packages/edition-slices/src/flexible-tiles/tile-vertical-a/index.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+interface Props {
+  onPress: () => void;
```
Comment (kamiranoff) [inline packages/edition-slices/src/flexible-tiles/tile-vertical-a/index.tsx:17]: shouldn't that be `onPress: (args: { id: string }) => void;` as defined in `tile-link`?
or  maybe `TileLinkProps['onPress']`

File: packages/edition-slices/__tests__/ios/tile-vertical-a.test.js (hunk 2 above)
Comment (c-j-j) [reply packages/edition-slices/__tests__/ios/tile-vertical-a.test.js:1]: Good spot... we don't need the .native.js extension here. 

However, we do have around 230 files with that extension. Is it worth creating a new PR that'll change all of them, including this? Could cause a few conflicts

Comment (aronshamash) [reply packages/edition-slices/__tests__/ios/tile-vertical-a.test.js:1]: I think that is for a separate PR to update all of them

File: packages/edition-slices/src/flexible-tiles/tile-vertical-a/styles.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import {
+  fonts,
+  spacing,
+  editionBreakpoints,
+  globalSpacingStyles,
+} from "@times-components-native/styleguide";
+
+const fontSizeResolver: Record<string, number> = {
```
Comment (kamiranoff) [inline packages/edition-slices/src/flexible-tiles/tile-vertical-a/styles.tsx:8]: I think the type could be improved with : 
`Record<keyof typeof editionBreakpoints, number>`

Not entirely sure

File: packages/edition-slices/src/flexible-tiles/tile-vertical-a/styles.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  [editionBreakpoints.medium]: 20,
+  [editionBreakpoints.wide]: 20,
+  [editionBreakpoints.huge]: 22,
+};
+
+export default (breakpoint: string) => ({
```
Comment (kamiranoff) [inline packages/edition-slices/src/flexible-tiles/tile-vertical-a/styles.tsx:14]: Just curious,
There is an argument against `default export`:
https://basarat.gitbook.io/typescript/main-1/defaultisbad

I think it is debatable depending on what the team think

File: packages/edition-slices/src/tiles/shared/tile-image.tsx [hunk 6]
```diff
@@ -1,16 +1,29 @@
...
+  relativeHeight: number;
+  relativeVerticalOffset: number;
+  relativeHorizontalOffset: number;
+  fill: boolean;
+  style: ViewStyle;
+  onLayout?: () => void;
+  uri: string;
+}
+
+const TileImage: FC<Props> = ({ hasVideo, style, onLayout, ...props }) => {
```
Comment (kamiranoff) [inline packages/edition-slices/src/tiles/shared/tile-image.tsx:20]: Also not related to this PR, but the prop drilling `...props`. I wonder if it wouldn't be better to be explicit about the props passed in.

**Review by kamiranoff — APPROVED** [review]

File: packages/edition-slices/src/flexible-tiles/tile-vertical-a/index.tsx (hunk 3 above)
Comment (c-j-j) [reply packages/edition-slices/src/flexible-tiles/tile-vertical-a/index.tsx:17]: Yes, good point. I think it ultimately calls this function here https://github.com/newsuk/times-components-native/blob/f0aba7c4fbbf30e7f66cf9c0d5d5ca0ce785515f/packages/pages/src/section/section.js#L27

So I think I'll create a shared type between these two.

**Review by aronshamash — APPROVED** [review]
just with the dir change. Lets tackle the other stuff as we go along

File: packages/edition-slices/src/flexible-tiles/tile-vertical-a/styles.tsx (hunk 5 above)
Comment (c-j-j) [reply packages/edition-slices/src/flexible-tiles/tile-vertical-a/styles.tsx:14]: I totally agree. I do prefer named exports, and definitely think they make a codebase easier to work with. However, this repo predominantly uses default exports, so it's going to be a big task to change/will introduce inconsistency

File: packages/edition-slices/src/tiles/shared/tile-image.tsx (hunk 6 above)
Comment (c-j-j) [reply packages/edition-slices/src/tiles/shared/tile-image.tsx:20]: I agree with the point, this was really just converting a js file to tsx to fix some typescript issues without any other changes. I'm happy to create a subsequent PR that defines what types the rest of those `props` have.

File: packages/edition-slices/src/flexible-tiles/tile-vertical-a/styles.tsx (hunk 4 above)
Comment (c-j-j) [reply packages/edition-slices/src/flexible-tiles/tile-vertical-a/styles.tsx:8]: A lot the styleguide code isn't currently typechecked, including the `eidtionBreakpoints` and where they're defined. Would be nice to do something like you've when that is converted over, and maybe creating a shared type as a lot of places have this breakpoint->style mapping

File: packages/edition-slices/src/tiles/shared/tile-image.tsx (hunk 6 above)
Comment (aronshamash) [reply packages/edition-slices/src/tiles/shared/tile-image.tsx:20]: definitely lets go with this approach in future PRs

**Pull request merged**